### PR TITLE
Update OpenShift workflow to use GHCR by default

### PR DESCRIPTION
**Pull request by tetchel**
- Simplifies required configuration since a registry account is now
  optional
- Update a variety of comments
- Use tools-installer to install oc
- Other small changes towards a better UX

Signed-off-by: Tim Etchells <tetchel@gmail.com>

## Pre-requisites

- [ ] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [ ] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [ ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [ ] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [ ] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [ ] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [ ] `name`: Name of the Code Scanning integration.
  - [ ] `organization`: Name of the organization producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [ ] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [ ] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [ ] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [ ] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [ ] Automation and CI workflows cannot be dependent on a paid service or product.

### CODE REVIEW

File: deployments/openshift.yml [hunk 1]
```diff
@@ -3,94 +3,99 @@
...
-# 5. Commit and push your changes.
-
-# For a more sophisticated example, see https://github.com/redhat-actions/spring-petclinic/blob/main/.github/workflows/petclinic-sample.yaml
-# Also see our GitHub organization, https://github.com/redhat-actions/
-# ▶️ See a video of how to set up this workflow at https://www.youtube.com/watch?v=6hgBO-1pKho
+# - Create an OpenShift app from the image and expose it to the internet
+
+# ℹ️ Before you begin:
+# 1. Have access to an OpenShift cluster. Refer to https://www.openshift.com/try
+# 2. The project you wish to add this workflow to should have a Dockerfile.
```
Comment (divyansh42) [inline deployments/openshift.yml:15]: By reading at this point users who don't have a Dockerfile might think that this workflow is not suitable for their use case.
Can we remove this requirement or at least mention that this workflow works without Dockerfile too.

File: deployments/openshift.yml [hunk 2]
```diff
@@ -3,94 +3,99 @@
...
+  APP_NAME: ""
+
+  # ✏️ EDIT with the port your application should be accessible on.
+  # If the container image exposes exactly one port, this can be left blank.
+  APP_PORT: ""
 
-  # ⬇️ EDIT to log into your OpenShift cluster and set up the context.
+  # ✏️ EDIT your repository secrets to log into your OpenShift cluster and set up the context.
   # See https://github.com/redhat-actions/oc-login#readme for how to retrieve these values.
   OPENSHIFT_SERVER: ${{ secrets.OPENSHIFT_SERVER }}
```
Comment (tetchel) [inline deployments/openshift.yml:43]: move these up

File: deployments/openshift.yml [hunk 3]
```diff
@@ -3,94 +3,99 @@
...
+# - Create an OpenShift app from the image and expose it to the internet
+
+# ℹ️ Before you begin:
+# 1. Have access to an OpenShift cluster. Refer to https://www.openshift.com/try
+# 2. The project you wish to add this workflow to should have a Dockerfile.
+#   - If you don't have a Dockerfile at the repository root,
+#     or if it is named differently (eg. 'Containerfile'), see the buildah-build step.
+#   - The buildah-build step can be replaced with any other image build step if you do not wish to use a Dockerfile.
+
+# ℹ️ This workflow requires some configuration.
```
Comment (tetchel) [inline deployments/openshift.yml:20]: Reword this - the configuration is now all optional
the secrets just have to be created
https://cli.github.com/manual/gh_secret_set

File: deployments/openshift.yml [hunk 4]
```diff
@@ -3,94 +3,99 @@
...
+
+  # ✏️ EDIT with the port your application should be accessible on.
+  # If the container image exposes exactly one port, this can be left blank.
+  APP_PORT: ""
 
-  # ⬇️ EDIT to log into your OpenShift cluster and set up the context.
+  # ✏️ EDIT your repository secrets to log into your OpenShift cluster and set up the context.
   # See https://github.com/redhat-actions/oc-login#readme for how to retrieve these values.
   OPENSHIFT_SERVER: ${{ secrets.OPENSHIFT_SERVER }}
   OPENSHIFT_TOKEN: ${{ secrets.OPENSHIFT_TOKEN }}
```
Comment (tetchel) [inline deployments/openshift.yml:44]: https://github.com/redhat-actions/oc-login/wiki/Using-a-Service-Account-for-GitHub-Actions

File: deployments/openshift.yml (hunk 1 above)
Comment (tetchel) [reply deployments/openshift.yml:15]: :+1:

**Pull request merged**